lib/facility_locations.rb

```diff
@@ -0,0 +1,134 @@
+require 'json'
+require './lib/facility'
+require './lib/dmv_data_service'
+
+=begin
+Since there are different formatting rules for each state,
+these first 3 blocks will determine which state API was passed
+in and call the method specific to that stae.
+A more direct solution would have been to just add a second
+parameter after 'facilities' in create_facilities and pass in an
+argument for the state you are working with. The spirit of the
+assignment seemed to encourage solving problems progamattically,
+thus my chosen solution.
+All addresses were formated to match those of the original
+interaction patterns.
+=end
+
+class FacilityLocations
+  def create_facilities(facilities)
+    # This block fetches the zipcode from the first element.
+    # The IF block is needed because the zipcode key is named
+    # differetly across APIs.
+    # I realize that there is the possibility that a state might have a
+    # bad or no zipcode for the first element. I did not code for this
+    # because I ran out of time. If I had another day, I would have included
+    # an IF block to test for nil OR wrong-length zipcode and then
+    # move to the next element.
+    if facilities.first.include?(:zip_code)
+      zipcode = facilities.first[:zip_code]
+    else
+      zipcode = facilities.first[:zipcode]
+    end
+
+# Call the appropriate method based on the state we are working with.
+    location = zipcode_locator(zipcode)
+    if location == "or"
+      or_facilities(facilities)
+    elsif location == "mo"
+      mo_facilities(facilities)
+    elsif location == "ny"
+      ny_facilities(facilities)
+    end
+  end
+  
+  # The zipcode_locator determines which state API to use.
+  def zipcode_locator(zipcode)
```

Comment:
What a great helper method! I wonder though, if we could just combine the zipcode_locator into the method above though, could just do `if zipcode.between?(97001, 97920)` then `or_facilities(facilities)`. Does that make sense? I don't think we're using the 'or', 'mo', and 'ny' for anything else, so could jsut combine the two steps. Really like your thoguht process on breaking up this logic!